File: php/char_frequency.php

<?php

// count occurence of characters in string

$string = $argv[1];

$char_counts = array();

foreach (str_split($string) as $value) {
    if (array_key_exists($value, $char_counts)) {
        $char_counts[$value] += 1;
    } else {
        $char_counts[$value] = 1;
    }
}

foreach ($char_counts as $key => $value) {
    echo $key . ":" . $value . "\n";
}

?>

File: lua/char_frequency.lua

-- count occurence of characters in string

str = arg[1]
str_len = string.len(str)
char_counts = {}

for i = 1, str_len, 1 do
	curr_char = string.sub(str, i, i)
	if char_counts[curr_char] == nil then
		char_counts[curr_char] = 1
	else
		char_counts[curr_char] = char_counts[curr_char] + 1
	end
end

for k, v in pairs(char_counts) do
	io.write(k .. ':' .. v .. '\n');
end